src/mmw/js/src/analyze/models.js

```diff
@@ -54,27 +54,37 @@ var AnalyzeTaskModel = coreModels.TaskModel.extend({
     }
 });
 
-AnalyzeTaskModel.getSingleton = function(App, aoi) {
-    if (!App.analyzeModel) {
-        App.analyzeModel = createTaskModel(JSON.stringify(aoi));
+var AnalyzeTaskCollection = Backbone.Collection.extend({
+    model: AnalyzeTaskModel
+});
+
+AnalyzeTaskCollection.getSingleton = function(App, aoi) {
+    if (!App.analyzeCollection) {
+        App.analyzeCollection = new AnalyzeTaskCollection([
+            { area_of_interest: aoi, taskName: 'analyze' },
+            { area_of_interest: aoi, taskName: 'analyze/animals' },
+            { area_of_interest: aoi, taskName: 'analyze/pointsource' },
+        ]);
     }
 
-    return App.analyzeModel;
+    return App.analyzeCollection;
 };
 
-// Pass in the serialized Area of Interest for
-// caching purposes (_.memoize returns the same
-// results for any object), and deserialize
-// the AoI for use on the model.
-var createTaskModel = _.memoize(function(aoi) {
-    return new AnalyzeTaskModel({
-        area_of_interest: JSON.parse(aoi)
-    });
-});
+function createAnalyzeResultViewModelCollection(App, aoi) {
```

Comment:
If I move this logic to the file that describes the `App` class, I will have to `require` `AnalyzeTaskCollection` to be able to call `getSingleton` from here, which could lead to cyclic dependencies.

If I define this on `App` here, by `require`ing `App` and then doing `App.createAnalyzeResultViewModelCollection = function(aoi)`, I think that is more confusing because then it is less clear where the function gets defined and attached.